src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,184 @@
+        final List<DetailAST> caseLabels = getAllCaseLabels(ast);
+        final boolean hasNullCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasLiteralNull);
+        if (!hasNullCaseLabel) {
+            final boolean hasPatternCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasPatternCaseLabel);
+            final boolean hasStringCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasStringCaseLabel);
+            if (hasPatternCaseLabel || hasStringCaseLabel) {
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * Gets all case labels in the given switch AST node.
+     *
+     * @param switchAST the AST node representing {@code LITERAL_SWITCH}
+     * @return a list of all case labels in the switch
+     */
+    private static List<DetailAST> getAllCaseLabels(DetailAST switchAST) {
+        final List<DetailAST> caseLabels = new ArrayList<>();
+        DetailAST ast = switchAST.getFirstChild();
+        while (ast != null) {
+            // case group token may have several LITERAL_CASE tokens
+            TokenUtil.forEachChild(ast, TokenTypes.LITERAL_CASE, caseLabels::add);
+            ast = ast.getNextSibling();
+        }
+        return Collections.unmodifiableList(caseLabels);
+    }
+
+    /**
+     * Checks if the given case AST node has a null label.
+     *
+     * @param caseAST the AST node representing {@code LITERAL_CASE}
+     * @return true if the case has {@code null} label, false otherwise
+     */
+    private static boolean hasLiteralNull(DetailAST caseAST) {
+        return Optional.ofNullable(caseAST.findFirstToken(TokenTypes.EXPR))
+                .map(exp -> exp.findFirstToken(TokenTypes.LITERAL_NULL))
+                .isPresent();
+    }
+
+    /**
+     * Checks if the given case AST node has a pattern variable declaration label
+     * or record pattern definition label.
+     *
+     * @param caseAST the AST node representing {@code LITERAL_CASE}
+     * @return true if case has a pattern in its label
+     */
+    private static boolean hasPatternCaseLabel(DetailAST caseAST) {
+        return caseAST.findFirstToken(TokenTypes.RECORD_PATTERN_DEF) != null
+               || caseAST.findFirstToken(TokenTypes.PATTERN_VARIABLE_DEF) != null
+               || caseAST.findFirstToken(TokenTypes.PATTERN_DEF) != null;
+    }
+
+    /**
+     * Checks if the given case contains a string in its label.
+     * It may contain a single string literal or a string literal
+     * in a concatenated expression.
+     *
+     * @param caseAST the AST node representing {@code LITERAL_CASE}
+     * @return true if switch block contains a string case label
+     */
+    private static boolean hasStringCaseLabel(DetailAST caseAST) {
+        DetailAST curNode = caseAST;
+        boolean hasStringCaseLabel = false;
+        while (curNode != null) {
```

Comment:
Since this is a new validation, I just wanted to verify you tried to ensure by debugging that we won't go beyond the current `caseAST`?

These type of mini-walkers have this issue where they go beyond where they start and look into too much. From glancing at the code, I am not seeing anything that stops this except a `null` AST or a string literal. I would expect it to end when it comes back to `caseAST` again (IE finishes the current case or it's case group, however we work this).

Reply:
Thanks for pointing this out. Yes I missed a condition to not go beyond the current case label expression 

```
        |       |               |   |--LITERAL_CASE -> case [99:12] <--- we should just iterate on this subtree
        |       |               |   |   `--EXPR -> EXPR [99:33]   
        |       |               |   |       `--PLUS -> + [99:33]
        |       |               |   |           |--PLUS -> + [99:27]
        |       |               |   |           |   |--PLUS -> + [99:23]
        |       |               |   |           |   |   |--PLUS -> + [99:19]
        |       |               |   |           |   |   |   |--NUM_INT -> 1 [99:17]
        |       |               |   |           |   |   |   `--NUM_INT -> 2 [99:21]
        |       |               |   |           |   |   `--NUM_INT -> 3 [99:25]
        |       |               |   |           |   `--STRING_LITERAL -> "a" [99:29]
        |       |               |   |           `--NUM_INT -> 5 [99:35]
        |       |               |   |--LAMBDA -> -> [99:37]    <--- once we reach  `->` or `:` it means that we finished the deepscan of case expr tree then we can break
```

Reply:
I assume this is done?

I am not really liking the `TokenUtil.isOfType` to call, but it seems necessary since our end AST is in different positions and some of the other tokens around it could be missing.

Reply:
> I assume this is done?

Yes, this is done. we can resolve it